### ISSUE TITLE
testStart callback does not wait on returned Promise.

### ISSUE THREAD

**Issue by jeberger** [qunitjs/qunit#1374]
## Tell us about your runtime:

* **QUnit version**: 2.7.1
* **What environment are you running QUnit in? (e.g., browser, Node)**: node v10.12.0
* **How are you running QUnit? (e.g., script, testem, Grunt)**: using the qunit script (eg. `./node_modules/qunit/bin/qunit`)

## What are you trying to do?

Code that reproduces the problem:

```js
"use strict";

QUnit.testStart (() => {
   console.log ("waiting");
   return new Promise ((resolve, reject) => setTimeout (resolve, 5000))
      .then (() => console.log ("proceeding"));
});
QUnit.testDone (() => console.log ("done"));

QUnit.test ("test", (assert) => {
   console.log ("inside the test");
   assert.equal (2+2, 4);
});
```

Command line:

```
./node_modules/qunit/bin/qunit test.js
```

## What did you expect to happen?

According to [the documentation](https://api.qunitjs.com/callbacks/QUnit.testStart), QUnit should wait for the promise before proceeding, so I expected to see the traces:

```
waiting
proceeding
inside the test
done
```

in that order with a 5s delay between `waiting` and `proceeding`.

## What actually happened?

The test proceeded immediately without waiting for the promise. I got the traces:

```
waiting
inside the test
done
proceeding
```

with the expected 5s delay before the `proceeding` trace, but by that time the test has already run.

**Comment by rwjblue** [comment]
> * **QUnit version**: 2.7.1

Per [History.md](https://github.com/qunitjs/qunit/blob/master/History.md#280--2018-11-02) supporting async in these callbacks was not landed until 2.8.0 (by way of #1307). 

Can you update and re-run your test?

**Comment by jeberger** [comment]
Upgrading QUnit fixes the issue. Thanks for the prompt response.
